Sources/PipecatClientIOSGeminiLiveWebSocket/util/AudioManager.swift

```diff
@@ -377,4 +313,106 @@ final class AudioManager {
             }
         }
     }
+    
+    // MARK: - Available Devices
+    
+    private func refreshAvailableDevices(suppressDelegateCallbacks: Bool = false) {
+        if !isManaging {
+            return
+        }
+        
+        // Check for change in available devices
+        let newAvailableDevices = getAvailableDevices()
+        if availableDevices != newAvailableDevices {
+            availableDevices = newAvailableDevices
+            if !suppressDelegateCallbacks {
+                delegate?.audioManagerDidChangeAvailableDevices(self)
+            }
+        }
+    }
+    
+    private func getCurrentAudioDevice() -> AudioDeviceType {
+        let defaultDevice: AudioDeviceType = Self.defaultDevice
+
+        guard let firstOutput = self.audioSession.currentRoute.outputs.first else {
+            return defaultDevice
+        }
+
+        guard let audioDevice = AudioDeviceType(sessionPort: firstOutput.portType) else {
+            return defaultDevice
+        }
+
+        return audioDevice
+    }
+    
+    // Adapted from WebrtcDevicesManager in Daily
+    private func getAvailableDevices() -> [Device] {
```

Comment:
I guess this logic has not changed, right ?

Reply:
Correct! Didn't want to touch this.